service/gcs/bridge/bridge.go

```diff
@@ -422,12 +403,10 @@ func (b *Bridge) createContainer(w ResponseWriter, r *Request) {
 	// CreateContainerInfo struct.
 	var settings prot.VMHostedContainerSettings
 	if err := commonutils.UnmarshalJSONWithHresult([]byte(request.ContainerConfig), &settings); err != nil {
```

Comment:
As a followup, it would be good to refine this request type to avoid needing multiple unmarshals. If you look again at the protocol.go in HCS, you can see how I defined types to work around the dumb JSON-in-string nesting.

I wonder if you can just take protocol.go.

Reply:
Yep agreed.